### ISSUE TITLE
Restore of an offline backup causes startup to fail

### ISSUE THREAD

**Issue by m-borger** [pgbackrest/pgbackrest#2018]
### Use case
Making offline backups of a bulk loading system between runs. For performance reasons archive_mode is set to off and wal_level to minimal. No PITR required, just a restore of the situation when the backup was taken.

### Configuration
pgbackrest.conf:
```ini
[dwh]
pg1-path=/pgdata/dwh
repo1-retention-full-type=count
repo1-retention-full=5
start-fast=y
```
postgresql.conf:
```ini
archive_mode = 'off'
wal_level = 'minimal'
max_wal_senders = 0
```

### The backup command
```sh
pg_ctl stop -D /pgdata/dwh && \
pgbackrest --stanza=dwh --type=full --no-online backup && \
pg_ctl start -D /pgdata/dwh
```

### The restore command
IMO two possibilities make sense here:
```sh
pgbackrest --stanza=dwh --set=20230303-081514F --type=none restore
# or
pgbackrest --stanza=dwh --set=20230303-081514F --type=immediate restore
```
### The problem
Both options place a `recovery.signal` at the end, causing the server to refuse to start with different, valid reasons:
- type=none:
`FATAL:  must specify restore_command when standby mode is not enabled`
- type=immediate, wal_level=minimal:
`FATAL:  WAL was generated with wal_level=minimal, cannot continue recovering`
- type=immediate, wal_level=replica:
`FATAL:  recovery ended before configured recovery target was reached`

### The solution
In all cases above, the problem can be resolved by manually removing `recovery.signal`.

### The question
This behavior could be very confusing for less experienced users. Wouldn't it be nice to either:
- Document this somewhere with the `--type` option of the restore command and/or `--no-online` option of the backup command?
- Leave out the `recovery.signal` placement when the backup was taken offline (maybe just for `type=none`)?
- Or did I miss something and could this situation be prevented otherwise?

**Comment by dwsteele** [comment]
You are the very first user that I can recall using this feature. We originally added it to aid in testing, though it was always intended to be a regular feature as well. It just turned out to never be used as far as we knew -- until now!

It looks like this was broken in 29e132f5e9282 when PG12 support was added. There should have been an exception to writing `recovery.signal` for offline backups.

We'll get this fixed. I think what we want to do here is default `type=none` when restoring an offline backup and omit `recovery.conf` only in that case. If an offline backup is made from cluster with wal_level > minimal then PITR is still possible so we don't want to exclude that possibility. If the user messes up they'll get an error like you did above. But, we want to give them one possible avenue to success (without needing to delete stuff) and we'll document that as well.

I suppose we could make this a little smarter by reading wal_level out of pg_control but that's probably overkill in this case.

Thoughts, @sfrost?

**Comment by dwsteele** [comment]
PR #2019 contains my proposed solution to this issue. Please have a look!

**Comment by m-borger** [comment]
First of all: thank you and your fellow contributors for all the effort making pgBackRest the valuable tool it is today.

Some thoughts: I might be the first user _using_ this feature, bringing this behavior to light. It could as well be, that more users than we think _rely_ on this feature somewhere in the future (when they need a restore and did not test it after the behavior changed). I think it is valuable to have this issue for reference and the fix in 2.45.
Glad I could help, have a nice weekend!

**Comment by dwsteele** [comment]
> First of all: thank you and your fellow contributors for all the effort making pgBackRest the valuable tool it is today.

You are welcome!

> It could as well be, that more users than we think rely on this feature somewhere in the future (when they need a restore and did not test it after the behavior changed). I think it is valuable to have this issue for reference and the fix in 2.45.

Completely agree. My statement was more about why this had not been caught before rather than questioning the usefulness of the feature.
